contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3657,10 +3663,19 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 									char *permissions[] = {"select", "insert", "update", "references", "delete"};
 									for(i = 0; i < 5; i++)
 									{
-										if ((rol_spec->rolename != NULL) && check_bbf_schema_for_entry(logical_schema, "ALL", permissions[i], rol_spec->rolename) && !has_schema_perms)
-											has_schema_perms = true;
-										if ((rol_spec->rolename != NULL) && check_bbf_schema_for_entry(logical_schema, obj, permissions[i], rol_spec->rolename))
-											del_from_bbf_schema(logical_schema, obj, permissions[i], rol_spec->rolename);
+										if ((rol_spec->rolename != NULL) && (strcmp(rol_spec->rolename, "public") != 0))
+										{
+											/*
+											 * 1. If only schema permission exists, don't revoke any permission.
+											 * 2. If only object permission exists, delete entry from the catalog and revoke permission.
+											 * 3. If both schema and object permission exist, don't revoke any permission but delete object
+											 *    entry from the catalog.
+											 */
+											if (check_bbf_schema_for_entry(logical_schema, "ALL", permissions[i], rol_spec->rolename) && !has_schema_perms)
```

Comment:
Are you saying that users wont be able to create and grant permissions to an object named as dbo.ALL?
This is severe!

Reply:
Also create Macros for such hardcoded values